src/parser/template/parameters.ts

```diff
@@ -1,155 +1,33 @@
-      )
-    );
-  }
-
-  public has(name: string) {
-    return !!this.parameters[name];
-  }
-
-  public parse(name: string, value: string) {
-    const param = this.parameters[name];
-    if (!param) {
-      throw new Error(`Unknown parameter: ${name}`);
-    }
-    return parseParamValue(name, param, value);
-  }
-}
-
-function parseParamValue(name: string, param: schema.Parameter, value: string) {
-  const ptype = parseParameterType(param.Type);
-
-  if (ptype.source === 'ssm') {
-    throw new Error('SSM parameter values not yet supported');
-  }
-
-  let ret: ContextValue;
-  if (ptype.type === 'list') {
-    ret = value.split(',').map((x) => parseScalar(x, ptype.elementType));
-  } else {
-    ret = parseScalar(value, ptype);
-  }
-
-  if (param.AllowedValues && !param.AllowedValues.includes(assertString(ret))) {
-    throw new Error(
-      `Parameter ${name} ${
-        param.ConstraintDescription ?? `not in list ${param.AllowedValues}`
-      }`
-    );
-  }
-
-  if (
-    param.AllowedPattern &&
-    !new RegExp(param.AllowedPattern).test(assertString(ret))
-  ) {
-    throw new Error(
-      `Parameter ${name} ${
-        param.ConstraintDescription ??
-        `does not match pattern ${param.AllowedPattern}`
-      }`
-    );
-  }
-
-  // @todo: More validations here
-
-  return ret;
-
-  function parseScalar(scalar: string, type: ScalarParameterType) {
-    switch (type.type) {
-      case 'number':
-        return parseNumber(scalar).asString;
-      case 'string':
-        return scalar;
-    }
-  }
-}
-
-function assertString(x: ContextValue): string {
-  if (typeof x !== 'string') {
-    throw new Error(`Expected string, got ${JSON.stringify(x)}`);
-  }
-  return x;
-}
-
-function parseParameterType(specifier: string): ParameterType {
-  let source: ParameterSource = { source: 'direct' };
-  const m = specifier.match(/^AWS::SSM::Parameter::Value<(.*)>$/);
-  if (m) {
-    source = { source: 'ssm' };
-    specifier = m[1];
-  }
-
-  if (specifier === 'CommaDelimitedList') {
-    specifier = 'List<String>';
-  }
-
-  let isList = false;
-  const mm = specifier.match(/^List<(.*)>$/);
-  if (mm) {
-    isList = true;
-    specifier = mm[1];
-  }
-
-  let type: ScalarParameterType;
-  switch (specifier) {
-    case 'String':
-      type = { type: 'string' };
-      break;
-    case 'Number':
-      type = { type: 'number' };
-      break;
-    default:
-      type = {
-        type: 'string',
-        resourceIdentifier: specifier as ResourceIdentifier,
-      };
-      break;
-  }
+export function parseParameter(x: unknown): TemplateParameter {
+  const param = assertObject(x);
 
   return {
-    ...source,
-    ...(isList ? { type: 'list', elementType: type } : type),
+    type: ifField(param, 'Type', assertString),
+    default: param.Default,
+    allowedPattern: ifField(param, 'AllowedPattern', assertString),
+    allowedValues: ifField(param, 'AllowedValues', assertList) as string[],
```

Comment:
I'm not sure about this. CDK expects a list of strings, but I cannot see this being an actual thing in the CFN spec. Due to the dynamic types in JS, this works and CDK currently passes values through.